GW2EIEvtcParser/EIData/CombatReplay/CombatReplay.cs

```diff
@@ -397,6 +397,39 @@ internal void AddTetherByEffectGUID(ParsedEvtcLog log, EffectEvent effect, strin
                 Decorations.Add(new LineDecoration(lifespan, color, new AgentConnector(effect.Dst), new AgentConnector(effect.Src)));
             }
         }
+
+        /// <summary>
+        /// Add tether decoration connecting a player to an agent.<br></br>
+        /// The <paramref name="buffId"/> is sourced by an agent that isn't the one to tether to.
+        /// </summary>
+        /// <param name="log">The log.</param>
+        /// <param name="player">The player to tether to <paramref name="toTetherAgentId"/>.</param>
+        /// <param name="buffId">ID of the buff sourced by <paramref name="buffSrcAgentId"/>.</param>
+        /// <param name="buffSrcAgentId">ID of the agent sourcing the <paramref name="buffId"/>. Either <see cref="ArcDPSEnums.TargetID"/> or <see cref="ArcDPSEnums.TrashID"/>.</param>
+        /// <param name="toTetherAgentId">ID of the agent to tether to the <paramref name="player"/>. Either <see cref="ArcDPSEnums.TargetID"/> or <see cref="ArcDPSEnums.TrashID"/>.</param>
+        /// <param name="color">Color of the tether.</param>
+        /// <param name="firstAwareThreshold">Time threshold in case the agent spawns before the buff application.</param>
+        internal void AddTetherByThirdPartySrcBuff(ParsedEvtcLog log, AbstractPlayer player, long buffId, int buffSrcAgentId, int toTetherAgentId, string color, int firstAwareThreshold = 2000)
+        {
+            var buffEvents = log.CombatData.GetBuffData(buffId).Where(x => x.To == player.AgentItem && x.CreditedBy.IsSpecies(buffSrcAgentId)).ToList();
+            var buffApplies = buffEvents.OfType<BuffApplyEvent>().ToList();
+            var buffRemoves = new HashSet<BuffRemoveManualEvent>(buffEvents.OfType<BuffRemoveManualEvent>());
+            var agentsToTether = log.AgentData.GetNPCsByID(toTetherAgentId).ToList();
+
+            foreach (BuffApplyEvent buffApply in buffApplies)
+            {
+                long removalTime = buffRemoves.Where(x => x.Time > buffApply.Time) != null ? buffRemoves.FirstOrDefault(x => x.Time > buffApply.Time).Time : log.FightData.LogEnd;
+                (long, long) lifespan = (buffApply.Time, removalTime);
+
+                foreach (AgentItem agent in agentsToTether)
+                {
+                    if ((Math.Abs(agent.FirstAware - buffApply.Time) < firstAwareThreshold || agent.FirstAware >= buffApply.Time) && agent.FirstAware < removalTime)
+                    {
+                        Decorations.Add(new LineDecoration(lifespan, color, new AgentConnector(agent), new AgentConnector(player)));
```

Comment:
Shouldn't there be a break here? To avoid iterating over the remaining agentsToTether when we have found what we are looking for.

Reply:
If i break there i turn this
![6457](https://github.com/baaron4/GW2-Elite-Insights-Parser/assets/23245218/873a84ac-a1fa-4d2e-9d24-3d91396ac1de)
into this
![6458](https://github.com/baaron4/GW2-Elite-Insights-Parser/assets/23245218/71fd4071-4f7d-4f0a-955e-1c581ebdc3cd)

but also i found this
![6456](https://github.com/baaron4/GW2-Elite-Insights-Parser/assets/23245218/398343bc-1c4d-4a44-ae51-2d4604734d1d)
had no idea this could be the case 🤔 it's on 20231206-230437